packages/components/src/components/hds/pagination/nav/arrow.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import {
+  HdsPaginationDirectionValues,
+  HdsPaginationDirectionAriaLabelValues,
+  HdsPaginationDirectionLabelValues,
+} from '../types.ts';
+
+import type { HdsIconSignature } from '../../icon/index.ts';
+import type { HdsInteractiveSignature } from '../../interactive';
+import type {
+  HdsPaginationDirections,
+  HdsPaginationDirectionAriaLabels,
+  HdsPaginationDirectionLabels,
+} from '../types';
+
+interface HdsPaginationControlArrowContent {
+  label: HdsPaginationDirectionLabels;
+  icon: HdsIconSignature['Args']['name'];
+  ariaLabel: HdsPaginationDirectionAriaLabels;
+}
+
+interface HdsPaginationControlArrowArgs {
+  direction: HdsPaginationDirections;
+  disabled?: boolean;
+  showLabel?: boolean;
+  onClick?: (direction: HdsPaginationDirections) => void;
+}
+
+interface HdsPaginationControlArrowSignature {
+  Args: HdsPaginationControlArrowArgs & HdsInteractiveSignature['Args'];
+  Element: HdsInteractiveSignature['Element'];
+}
+
+export const DIRECTIONS: HdsPaginationDirections[] = [
+  HdsPaginationDirectionValues.Prev,
+  HdsPaginationDirectionValues.Next,
+];
+
+export default class HdsPaginationControlArrowComponent extends Component<HdsPaginationControlArrowSignature> {
+  get content(): HdsPaginationControlArrowContent {
+    const { direction } = this.args;
+
+    assert(
+      `@direction for "Pagination::Nav::Arrow" must be one of the following: ${DIRECTIONS.join(
+        ', '
+      )}; received: ${direction}`,
+      DIRECTIONS.includes(direction)
+    );
+
+    let content: Partial<HdsPaginationControlArrowContent> = {};
+    if (direction === HdsPaginationDirectionValues.Prev) {
+      content = {
+        label: HdsPaginationDirectionLabelValues.Prev,
+        icon: 'chevron-left',
+        ariaLabel: HdsPaginationDirectionAriaLabelValues.Prev,
+      };
+    }
+    if (direction === HdsPaginationDirectionValues.Next) {
+      content = {
+        label: HdsPaginationDirectionLabelValues.Next,
+        icon: 'chevron-right',
+        ariaLabel: HdsPaginationDirectionAriaLabelValues.Next,
+      };
+    }
+
+    return content as HdsPaginationControlArrowContent;
```

Comment:
Could this be refactored so we don't need the `return content as HdsPaginationControlArrowContent`? Especially since direction should always be defined and there's only 2 options?

```typescript
    if (direction === HdsPaginationDirectionValues.Prev) {
      return {
        label: HdsPaginationDirectionLabelValues.Prev,
        icon: 'chevron-left',
        ariaLabel: HdsPaginationDirectionAriaLabelValues.Prev,
      };
    } else {
      return {
        label: HdsPaginationDirectionLabelValues.Next,
        icon: 'chevron-right',
        ariaLabel: HdsPaginationDirectionAriaLabelValues.Next,
      };
    }
```

Reply:
That's a good point. I took a look and came up with an alternative. Thanks for the suggestion.